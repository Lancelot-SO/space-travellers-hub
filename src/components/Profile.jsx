import React from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import ListGroup from 'react-bootstrap/ListGroup';
import { useSelector } from 'react-redux';
import styles from './Profile.module.css';
import { selectReservedRockets } from '../redux/rocket/rocketSlice';

function Profile() {
  const missions = useSelector((state) => state.missions.missionstore);
  const reservedRockets = useSelector(selectReservedRockets);
  return (
    <div className="px-2 pr-2">
      <Row>
        <Col
          sm={12}
          lg={6}
          md={6}
          className="ps-lg-5"
        >
          <h2>My Missions</h2>
          <ListGroup>
            {missions.filter((mission) => mission.reserved === true).map((mission) => (
              <ListGroup.Item key={mission.mission_id}>
                {mission.mission_name}
              </ListGroup.Item>
            ))}
          </ListGroup>
        </Col>
        <Col
          sm={12}
          lg={6}
          md={6}
          className="pe-lg-5"
        >
          <h2>My Rockets</h2>
          <section className={styles.section}>
            <ul className={styles.list}>
              {reservedRockets.map((e) => (
                <li key={e.rocket_id} className={styles.listItem}>
                  {e.rocket_name}
                </li>
              ))}
            </ul>
          </section>

        </Col>

      </Row>
    </div>
  );
}

export default Profile;
