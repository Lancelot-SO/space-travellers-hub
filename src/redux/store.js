import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { createLogger } from 'redux-logger';
import rocketReducer from './rocket/rocketSlice';
import missionsReducer from './missions/missionsSlice';

const logger = createLogger({
  predicate: (getState, action) => action.type !== 'SOME_ACTION',
  collapsed: true,
});

const store = configureStore({
  reducer: combineReducers({
    rockets: rocketReducer,
    missions: missionsReducer,
  }),
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
