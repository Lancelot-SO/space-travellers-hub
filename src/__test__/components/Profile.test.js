import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Profile from '../../components/Profile';
import store from '../../redux/store';

describe('Profile', () => {
  it('renders correctly for reserved state', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
