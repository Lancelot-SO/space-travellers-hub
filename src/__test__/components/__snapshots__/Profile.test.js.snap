// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly for reserved state 1`] = `
<div
  className="px-2 pr-2"
>
  <div
    className="row"
  >
    <div
      className="ps-lg-5 col-lg-6 col-md-6 col-sm-12"
    >
      <h2>
        My Missions
      </h2>
      <div
        className="list-group"
        onKeyDown={[Function]}
      />
    </div>
    <div
      className="pe-lg-5 col-lg-6 col-md-6 col-sm-12"
    >
      <h2>
        My Rockets
      </h2>
      <section
        className="section"
      >
        <ul
          className="list"
        />
      </section>
    </div>
  </div>
</div>
`;
